sound/soc/sof/intel/hda-pcm.c

```
@@ -263,6 +263,18 @@ int hda_dsp_pcm_open(struct snd_sof_dev *sdev,
 		snd_pcm_hw_constraint_mask64(substream->runtime, SNDRV_PCM_HW_PARAM_FORMAT,
 					     SNDRV_PCM_FMTBIT_S16 | SNDRV_PCM_FMTBIT_S32);
 
+	/*
+	 * Set constraint on the period time to make sure that it is larger than
+	 * the DSP buffer size. This is the maximum DMA burst size that can
+	 * happen on stream start for example.
+	 * The period size must not be smaller than this.
+	 */
+	if (spcm->stream[substream->stream].dsp_buffer_time_ms)
+		snd_pcm_hw_constraint_minmax(substream->runtime,
+			SNDRV_PCM_HW_PARAM_PERIOD_TIME,
+			spcm->stream[substream->stream].dsp_buffer_time_ms * USEC_PER_MSEC,
```

I was wondering if this can be cause unexpected isssues with large values (e.g. a 100ms buffer) with appes that want a specific amount of periods. But yeah, I think in all cases I thought through, it's better to fail early. On the other hand, I can see a lot more confusion with apps that set the period size to low and expect period wakeups to come at a steady rate.

---

Right, that is true. On the other hand those apps will have other problems if they want to use small periods when the DMA burst in a different magnitude...